SingleCellDataExtraction.py

```diff
@@ -73,6 +74,22 @@ def MaskIDs(mask, mask_props=None):
 
     return dat
 
+def n_channels(image):
+    """Returns the number of channel in the input image. Supports [OME]TIFF and HDF5."""
+
+    image_path = Path(image)
+
+    if image_path.suffix in ['.tiff', '.tif', '.btf']:
+        s = tifffile.TiffFile(image).series[0]
+        return s.shape[0] if len(s.shape) > 2 else 1	# Handle (w,h) cases
```

Comment:
There are two edge cases that I'm aware of, one is that `s` has more than 3 dimensions, I was handling it [using an `assert` in my PR](https://github.com/labsyspharm/quantification/pull/31/files#diff-88f4952deede79e0d8f0fc5ba6f81a3b30f4ce02825077f85278643d3747daf1R176-R178). The other case can be reproduced with this

```python
import tifffile, numpy

# RGB(A) image in [Y, X, C, (A)] order 
img = numpy.dstack([numpy.eye(10)]*3) # img = numpy.dstack([numpy.eye(10)]*4)

tifffile.imsave('color.tif', img)
print(tifffile.TiffFile('color.tif').series[0].shape)
# >> (10, 10, 3)
```

and downstream code, such as reading one channel at a time, needs to be modified to handle that.

Reply:
Thanks, @Yu-AnChen. Having an `assert` is a great idea. Would `image_loaded_z = tifffile.imread(image, key=z)` you suggested below correctly slice the image array in the second case?